feat(config): `platformOptions` (#30582)
renovatebot/renovate: docs/usage/configuration-options.md

When `platformCommit` is enabled, Renovate will create commits with GitHub's API instead of using `git` directly.
This way Renovate can use GitHub's [Commit signing support for bots and other GitHub Apps](https://github.blog/2019-08-15-commit-signing-support-for-bots-and-other-github-apps/) feature.

## platformOptions

> Use this option to configure the platform specific configuration options provided by renovate.

Reviewer: ```suggestion
Use `platformOptions` to set platform-specific configuration options provided by Renovate.
```

Example:

```json
{
  "platformOptions": {
    "gitLabIgnoreApprovals": true,
    "platformVersion": "8.0.0"
  }
}
```

<!-- prettier-ignore -->
!!! note

>     The supported options can be of self-hosted or repo config type, make sure to use the sub-options in correct config.

Reviewer: ```suggestion
    There are two kinds of config options: self-hosted config option or repository config options. Make sure you're using the correct kind. And check if it's in the correct location in the config.
```

> Currently the following options are available:

Reviewer: ```suggestion
You can use these options:
```

Self-hosted config options:

> 1. [platformVersion](./self-hosted-configuration.md#platformversion)

Reviewer: ```suggestion
1. [`platformVersion`](./self-hosted-configuration.md#platformversion)
```

Repository config options:

### gitLabIgnoreApprovals

> Ignore the default project level approval(s), so that Renovate bot can automerge its merge requests, without needing approval(s).

Reviewer: ```suggestion
Ignore the default project level approval(s).
This lets Renovate automerge its Merge Requests, without needing approval(s).
```

> Under the hood, it creates a MR-level approval rule where `approvals_required` is set to `0`.

Reviewer: ```suggestion
How it works: Renovate creates a MR-level approval rule where `approvals_required` is set to `0`.
```

> This option works only when `automerge=true` and either `automergeType=pr` or `automergeType=branch`.

Reviewer: ```suggestion
This option only works when:

- `automerge=true`
- and `automergeType=pr` _or_ `automergeType=branch`
```

When `platformCommit` is enabled, Renovate will create commits with GitHub's API instead of using `git` directly.
This way Renovate can use GitHub's [Commit signing support for bots and other GitHub Apps](https://github.blog/2019-08-15-commit-signing-support-for-bots-and-other-github-apps/) feature.

> ## platformOptions

Reviewer: The latest build has test failures. I think you need to:

- Add more tests?
- Update the Jest snapshots?

Author: Yes, fixes otw
